collated/main/A0139661Y.md

```diff
@@ -0,0 +1,1790 @@
+    }
+
+    @Override
+    public void stop() {
+        logger.info("============================ [ Stopping CMDo ] =============================");
+        ui.stop();
+        try {
+            storage.saveUserPrefs(userPrefs);
+        } catch (IOException e) {
+            logger.severe("Failed to save preferences " + StringUtil.getDetails(e));
+        }
+        Platform.exit();
+        System.exit(0);
+    }
+    
+    @Subscribe
+    public void handleExitAppRequestEvent(ExitAppRequestEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        this.stop();
+    }
+    
+    /**
+     * Handles an event where the storage file has been changed.
+     * @param event
+     * 
+```
+###### /java/seedu/cmdo/MainApp.java
+``` java
+     */
+    @Subscribe
+    public void handleStorageFileChangedEvent(StorageFileChangedEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        try {
+        	syncUserPrefsToConfig();
+        } catch (Exception e) {
+        	logger.severe("Failed to update config file.");
+        }
+    }
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+}
+```
+###### /java/seedu/cmdo/model/ModelManager.java
+``` java
+        this.undoer = Undoer.getInstance();
+        logger.info("Saved new toDoList into Undoer stack. " + toDoList.toString());
+    }
+
+    public ModelManager() {
+        this(new ToDoList(), new UserPrefs());
+    }
+
+    public ModelManager(ReadOnlyToDoList initialData, UserPrefs userPrefs) {
+        toDoList = new ToDoList(initialData);
+        filteredTasks = new FilteredList<>(toDoList.getTasks());
+        this.userPrefs = userPrefs;
+
+```
+###### /java/seedu/cmdo/model/ModelManager.java
+``` java
+        this.undoer = Undoer.getInstance();
+        logger.info("Saved last stable toDoList into Undoer stack. " + toDoList.toString());
+    }
+
+    @Override
+    public void resetData(ReadOnlyToDoList newData) {
+    	toDoList.resetData(newData);
+        indicateToDoListChanged();
+    }
+
+    @Override
+    public ReadOnlyToDoList getToDoList() {
+        return toDoList;
+    }
+
+    /** Raises an event to indicate the model has changed */
+    private void indicateToDoListChanged() {
+        raise(new ToDoListChangedEvent(toDoList));
+    }
+    
+    /**
+     * Undo functionality
+     * 
+```
+###### /java/seedu/cmdo/model/ModelManager.java
+``` java
+     */
+    @Override
+    public synchronized void undo() throws CannotUndoException {
+    	try {
+    		ToDoList currentState = new ToDoList(toDoList);
+    		toDoList.resetData(undoer.undo(currentState));
+    		logger.info("Undo operation called.");
+    	} catch (EmptyStackException ese) {
+    		throw new CannotUndoException("Nothing to undo.");
+    	}
+    	indicateToDoListChanged();
+    	updateFilteredListToShowAll();
+    }
+    
+    /**
+     * Redo functionality
+     * 
+```
+###### /java/seedu/cmdo/model/ModelManager.java
```

Comment:
are all these functions suppose to be commented away? if yes why is there such a large chunk commented away